src/opm/input/eclipse/EclipseState/Grid/MULTREGTScanner.cpp

```diff
@@ -306,11 +305,8 @@ namespace Opm {
             const int regionId1 = region_data[globalIndex1];
             const int regionId2 = region_data[globalIndex2];
 
-            auto regPairPos = regMap.find({ regionId1, regionId2 });
-            if ((regionId1 != regionId2) && !regPairFound(regMap, regPairPos)) {
-                // 1 -> 2 not found.  Try reverse direction.
-                regPairPos = regMap.find({ regionId2, regionId1 });
-            }
+            auto regPairPos = (regionId1 <= regionId2) ? regMap.find({ regionId1, regionId2 })
+                : regMap.find({ regionId2, regionId1 });
```

Comment:
You could probably apply the same simplification in [`getRegionMultiplierNNC()`](https://github.com/OPM/opm-common/blob/2e70b89731aefd1ebe916bfb9fad07daa7df0bd8/src/opm/input/eclipse/EclipseState/Grid/MULTREGTScanner.cpp#L356-L365) too.

Reply:
Nice catch. I am glad you found this and did not merge.

I actually think that the original PR introduced a bug for the case where one region is defaulted. Let's assume the other region is r, then the map would have an entry with pair `(r, r)` and the check for inequality of the two regions was vital.

It is  a bit concerning that this triggered no test failure. Maybe I am missing something.

Anyway the last commit makes sure that there is now key with a pair of equal numbers in the map for defaulted region indices and modifies [getRegionMultiplierNNC()](https://github.com/OPM/opm-common/blob/2e70b89731aefd1ebe916bfb9fad07daa7df0bd8/src/opm/input/eclipse/EclipseState/Grid/MULTREGTScanner.cpp#L356-L365) as proposed.

Reply:
I will add an additional test, so please hold off merging until then,